adr/0016-compliance-in-pepr.md

```diff
@@ -0,0 +1,43 @@
+# 14. Compliance in Pepr 
+
+Date: 2025-02-13
+
+
+## Status
+
+Pending
+
+
+## Context
+
+Pepr needs to generate reports for compliance purposes, demonstrating adherence to required security and compliance standards through specific bindings. To achieve this, it’s essential to map compliance controls to Pepr’s bindings. During our discovery process, we determined that Pepr’s role in compliance should focus on reporting rather than enforcement. This is because compliance controls are already enforced through the bindings themselves, and Pepr’s primary function is to report on the compliance status of resources.
+
+## Decision
+
+Extend the fluent API with a new keyword function that accepts a strongly typed control, allowing it to be associated with a given binding. This enables users to map compliance controls to Pepr’s bindings, facilitating the generation of compliance reports based on control status.
+
+
+### Consequences ###
+
+- Extend the Capability class with a new function, such as `CompliesWith()`, that accepts a strongly typed control to associate it with a given binding.
+- Store compliance data in `PeprStore` in a new instance prevent corruption of compliance data through Store API.
+- Enhance the Pepr CLI to generate compliance reports based on stored compliance data.
+
+##### Pros
+
+- Gives Pepr the ability to generate compliance reports based on the status of compliance controls on demand.
+- Straightforward to implement and maintain.
+- Should not make a major impact in terms of resource utilization.
+- Report generation can be automated.
+
+##### Cons
+
+- Module Authors need to update their modules to include compliance controls.
+- It is possible that Pepr may need to introduce a new depedency to support reporting in certain formats.
+
+### Related Repos ###
+
+These repos are related to this ADR in that they house collections of admission policies.
+
+- [Kubernetes Validating Admission Policy Library](https://github.com/vap-library/vap-library)
+- [CEL Admission Library](https://github.com/kubescape/cel-admission-library)
```

Comment:
A few missing bits that we discussed are:
- pepr should only provide the mechanism to run the validation, the validation rules can be bundled separately, potentially we'll create a default list of validation, but users can add their own;
- the report generation should be either on-demand, or on schedule.

Reply:
Thanks, I attempted to speak to this in the new `## Goals` section. I am not entirely sure that I understood the top point. Let me know how I can change it to account for you point if it does not meet it. I am guessing you are talking about a canned set of controls?